environment/simulation/simulation.py

```diff
@@ -27,60 +27,67 @@ def __init__(self,
                  gravity=(0.0, -900),
                  map_bottom_y_threshold=-300,
                  map_width=3
                  ):
-        self._space = None
-        self.gravity = gravity
-        self._map = []
-        self.map_width = map_width
-        self.map_bottom_y_threshold = map_bottom_y_threshold
+        # external simulation properties
+        self.debug = False
+        self.number_of_clusters = number_of_clusters
+        self.number_of_bots_per_cluster = number_of_bots_per_cluster
+        self.enemy_speed = enemy_speed
         self.initial_object_position = initial_object_position
-        self._map_segment_size = map_segment_size
-        map_initial_x_offset = (self.initial_object_position[0]-self._map_segment_size[0])
-        self._map_beginning = (-map_initial_x_offset, 0.0)
-        self._map_middle_right_boundary = (map_initial_x_offset, 0.0)
         self.ticks_per_step = ticks_per_step
         self.ticks_per_render_frame = ticks_per_render_frame
         self.difficulty = difficulty
-        self._giant_fry = None
+        self.map_segment_size = map_segment_size
+        self.map_width = map_width
+        self.map_bottom_y_threshold = map_bottom_y_threshold
+        self.gravity = gravity
+        self.screen_size = screen_size
+
+        # internal simulation properties
         self._simulation_is_running = True
-        self.number_of_clusters = number_of_clusters
-        self.number_of_bots_per_cluster = number_of_bots_per_cluster
-        self.enemy_speed = enemy_speed
+        self._dt = 1 / 60.0
+        map_initial_x_offset = (self.initial_object_position[0] - self.map_segment_size[0])
+        self._map_beginning = (-map_initial_x_offset, 0.0)
+        self._map_middle_right_boundary = (map_initial_x_offset, 0.0)
+        self._enemy_position = -map_initial_x_offset
+
+        # internal simulation objects
+        self._map = []
+        self._space = None
+        self._enemy = None
         self._clusters = None
         self._goal_object = None
-        self._thresholds = []
-        self._giant_fry_position = -map_initial_x_offset
-        self._dt = dt
 
         # pygame constants
-        self.screen_size = screen_size
         self._screen = pygame.display.set_mode(self.screen_size)
         self._clock = pygame.time.Clock()
         self._draw_options = pymunk.pygame_util.DrawOptions(self._screen)
 
         pygame.init()
 
-    def update_thresholds(self, *accelerations):
-        if len(accelerations) == len(self._thresholds):
-            # TODO: update velocity of all thresholds
-            pass
+    # input
+    def update_threshold(self, index, position):
+        self._clusters[index].threshold.position = position
+
+    # output
+    def threshold_positions(self):
+        return [cluster.threshold.position for cluster in self._clusters]
+
+    # output
+    def space_near_goal_object(self, window_size):
+        return pygame.image.tostring(self._screen, "RGB")
```

Comment:
a to nie trzeba zrobic najpierw redraw zeby na screenie zaktualizowaly sie pozycje obiektów?

Reply:
to juz nie kwestia pobrania wartosci funkcji w envie w odpowiednim momencie?

Reply:
ja mialam zrobic tylko wtyczki :) a Karol sobie odswieza wszystko, tak zrozumialam